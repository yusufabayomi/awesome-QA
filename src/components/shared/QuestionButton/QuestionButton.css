.question-btn {
  background-color: transparent;
  border: 0;
  color: #ffffff;
  font-size: 12px;
  padding: 0;
  margin: 0 5px;
}

.question-btn:hover {
  color: #a89fca;
  border-bottom: 1px dotted #a89fca;
}
