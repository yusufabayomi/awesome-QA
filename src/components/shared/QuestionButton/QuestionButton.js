import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './QuestionButton.css';

const propTypes = {
  buttonText: PropTypes.string.isRequired,
  emitEvent: PropTypes.func.isRequired,
};

const defaultProps = {
  icon: null,
};

const QuestionButton = ({ buttonText, icon, emitEvent }) => {
  const onClickButton = (event) => {
    event.stopPropagation();
    if (emitEvent) {
      emitEvent();
    }
  };
  return (
    <button data-testid='button' className='question-btn' onClick={(event) => onClickButton(event)}>
      {icon && <FontAwesomeIcon icon={icon} data-testid='buttonIcon' />} {buttonText}
    </button>
  );
};

QuestionButton.propTypes = propTypes;
QuestionButton.defaultProps = defaultProps;
export default QuestionButton;
