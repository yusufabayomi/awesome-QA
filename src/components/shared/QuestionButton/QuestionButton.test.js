import React from 'react';
import QuestionButton from './QuestionButton';
import { render, cleanup, fireEvent, createEvent } from '@testing-library/react';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons';

afterEach(cleanup);

describe('Renders Button Component', () => {
  const props = {
    buttonText: 'Edit',
    emitEvent: jest.fn(),
  };

  it('renders correctly', () => {
    const { asFragment } = render(<QuestionButton {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a FontAwesomeIcon when icon prop is defined', () => {
    const { getByTestId } = render(<QuestionButton {...props} icon={faPencilAlt} />);
    expect(getByTestId('buttonIcon')).toBeTruthy();
  });

  it('should stop event propagation on click event', () => {
    const { getByTestId } = render(<QuestionButton {...props} />);
    const questionButton = getByTestId('button');
    const buttonEvent = createEvent.click(questionButton);
    fireEvent.click(questionButton, buttonEvent);
    expect(buttonEvent.stopPropagation).toBeTruthy();
  });

  it('should emit a callback on click event', () => {
    const { getByTestId } = render(<QuestionButton {...props} />);
    fireEvent.click(getByTestId('button'));
    expect(props.emitEvent).toHaveBeenCalled();
  });
});
