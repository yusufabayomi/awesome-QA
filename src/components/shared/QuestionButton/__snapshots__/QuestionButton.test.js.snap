// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Button Component renders correctly 1`] = `
<DocumentFragment>
  <button
    class="question-btn"
    data-testid="button"
  >
     Edit
  </button>
</DocumentFragment>
`;
